packages/react-reconciler/src/__tests__/ReactSuspenseList-test.js

```diff
@@ -2567,4 +2569,170 @@ describe('ReactSuspenseList', () => {
+            </Suspense>
+            {addRow ? (
+              <Suspense fallback={<Fallback />}>
+                <Sleep time={12}>
+                  <D />
+                </Sleep>
+              </Suspense>
+            ) : null}
+          </SuspenseList>
+        </Profiler>
+      );
+    }
+
+    ReactNoop.render(<App suspendTail={true} />);
+
+    expect(Scheduler).toFlushAndYield([
+      'App',
+      'A',
+      'B',
+      'Suspend! [C]',
+      'Fallback',
+    ]);
+    expect(ReactNoop).toMatchRenderedOutput(
+      <>
+        <span>A</span>
+        <span>B</span>
+        <span>Loading...</span>
+      </>,
+    );
+    expect(onRender).toHaveBeenCalledTimes(1);
+
+    // The treeBaseDuration should be the time to render each child. The last
+    // one counts the fallback time.
+    // The actualDuration should also include the 5ms spent rendering the
+    // last suspended row.
+
+    // actualDuration
+    expect(onRender.mock.calls[0][2]).toBe(1 + 4 + 5 + 3);
+    // treeBaseDuration
+    expect(onRender.mock.calls[0][3]).toBe(1 + 4 + 3);
+
+    ReactNoop.render(<App suspendTail={false} />);
+
+    expect(Scheduler).toFlushAndYield(['App', 'A', 'B', 'C']);
+
+    expect(ReactNoop).toMatchRenderedOutput(
+      <>
+        <span>A</span>
+        <span>B</span>
+        <span>C</span>
+      </>,
+    );
+    expect(onRender).toHaveBeenCalledTimes(2);
+
+    // actualDuration
+    expect(onRender.mock.calls[1][2]).toBe(1 + 4 + 5);
+    // treeBaseDuration
+    expect(onRender.mock.calls[1][3]).toBe(1 + 4 + 5);
+
+    ReactNoop.render(<App addRow={true} suspendTail={true} />);
+
+    expect(Scheduler).toFlushAndYield([
+      'App',
+      'A',
+      'B',
+      'Suspend! [C]',
+      'Fallback',
+      // We rendered in together mode for the head, now we re-render with forced suspense.
+      'A',
+      'B',
+      'Suspend! [C]',
+      'Fallback',
+      // Lastly we render the tail.
+      'Fallback',
+    ]);
+
+    // Flush suspended time.
+    jest.advanceTimersByTime(1000);
+
+    expect(ReactNoop).toMatchRenderedOutput(
+      <>
+        <span>A</span>
+        <span>B</span>
+        <span hidden={true}>C</span>
+        <span>Loading...</span>
+        <span>Loading...</span>
+      </>,
+    );
+    expect(onRender).toHaveBeenCalledTimes(3);
+
+    // The treeBaseDuration should be the time to render the first two
+    // children and then two fallbacks.
+    // The actualDuration should also include rendering the content of
+    // the first fallback, as well as the second pass to render the head
+    // with force fallback mode.
+
+    // actualDuration
+    expect(onRender.mock.calls[2][2]).toBe((1 + 4 + 5 + 3) * 2 + 3);
+    // treeBaseDuration
+    expect(onRender.mock.calls[2][3]).toBe(
+      1 +
+        4 +
+        3 +
+        3 +
+        /* Resuspending a boundary also includes the content in base duration but it shouldn't */ 5,
```

Comment:
@bvaughn Like I mentioned in chat, I think this might be a bug or at least inconsistent. Seems like base duration includes both for updates (resuspending) but not for initial renders that suspend.

Reply:
Right, it seems not ideal that we count the hidden/suspended branch in the tree duration. It also makes sense *why* we do this though, because of how we accumulate child times.

I think to fix this we would need to make `resetChildExpirationTime` smart enough to not count both branches of `Suspense` component's children.

Want me to try to tackle this with a follow up?